### PR TITLE
Assessment Registry - Score form

### PR DESCRIPTION
**Pull request by shreeyash07**
- Addresses #2723 
- Depends on [feature/summary-score-cna](https://github.com/the-deep/server/pull/1324)

## Changes

* Add score form

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/gqlFragments.ts [hunk 1]
```diff
@@ -352,10 +352,24 @@ export const ASSESSMENT_REGISTRY_FRAGMENT = gql`
...
             clientId
             answer
+            clientId
             question {
                 id
                 question
             }
         }
+        scoreRatings {
+            scoreType
```
Comment (AdityaKhatri) [inline app/gqlFragments.ts:379]: id might be required, check with @Rup-Narayan-Rajbanshi

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        sectorsOptions?.enumValues,
+        (k) => k.name,
+        (item) => item.description ?? item.name,
+    );
+
+    const analyticalDensityValueIndex = listToMap(
+        analyticalDensityValue,
+        (k) => k.sector ?? '',
+        (_, __, index) => index,
+    );
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/index.tsx]: use useMemo

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
Also update scoreRating formSchema.

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/AnalyticalDensityInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    const error = getErrorObject(riskyError);
+
+    const [
+        analysisLevelCoveredOptions,
+        figureProvidedOptions,
+    ] = useMemo(() => ([
+        options?.analysisLevelCoveredOptions,
+        options?.figureProvidedOptions,
+    ]), [options]);
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/AnalyticalDensityInput/index.tsx]: useMemo is not required here.

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        analyticalDensityValue,
+        (k) => k.sector ?? '',
+        (_, __, index) => index,
+    ), [analyticalDensityValue]);
+
+    if (isNotDefined(value.sectors) || value.sectors.length <= 0) {
+        return (
+            <div>
+                Please select Sector from Focus tab
+            </div>
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/index.tsx]: Use Message component

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                    sector as AssessmentRegistrySectorTypeEnum
+                ];
+                const analyticalValue = isDefined(analyticalIndex)
+                    ? analyticalDensityValue?.[analyticalIndex]
+                    : undefined;
+                const selectedClientId = isDefined(analyticalIndex)
+                    ? analyticalDensityValue?.[analyticalIndex].clientId : undefined;
+                const analyticalError = isDefined(selectedClientId)
+                    ? analyticalDensityError?.[selectedClientId]
+                    : undefined;
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/AnalyticalDensityForm/index.tsx]: use ListVIew

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/QualityScoreForm/index.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        riskyError,
+    );
+
+    const scoreValue = value.scoreRatings;
+    const scoreError = getErrorObject(error?.scoreRatings);
+    const scoreValueIndex = listToMap(
+        scoreValue,
+        (k) => k.scoreType ?? '',
+        (_, __, index) => index,
+    );
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/QualityScoreForm/index.tsx]: use useMemo

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/QualityScoreForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    const groupScoreInputParams = useCallback(
+        (key: string): Omit<ScoreHeadingProps, 'children' | 'className'> => ({
+            analyticalState: key,
+        }),
+        [],
+    );
+
+    return (
+        <div className={styles.qualityScoreForm}>
+            <ListView
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/QualityScoreForm/index.tsx]: double nesting not required

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+            },
+        },
+    );
+
+    const scoreOptions = qualityScoreList?.project?.assessmentRegistryOptions?.scoreOptions;
+    const scoreToScoreGroupMap = listToMap(
+        scoreOptions,
+        (k) => k.scoreCriteria,
+        (d) => d.analyticalStatement,
+    );
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx]: move this inside the useMemo below

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+                const figureProvideValue = density.figureProvided?.length ?? 0;
+                const analysisLevelValue = density.analysisLevelCovered?.length ?? 0;
+                return (figureProvideValue * analysisLevelValue) / 10;
+            },
+        ).filter(isDefined);
+
+        const analyticalDensityValue = {
+            key: 'ANALYTICAL_DENSITY',
+            label: 'Analytical density',
+            score: median(sectorWiseDensityValue ?? []) ?? 0,
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx:129]:
```suggestion
            score: median(sectorWiseDensityValue ?? []),
```

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+
+                return acc;
+            },
+            {},
+        );
+
+        const scoreGroupScores = Object.values(scoreStatsList ?? []).sort(
+            (a, b) => (compareString(a.label, b.label)),
+        );
+
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx]: fix sorting

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+    return (
+        <div className={styles.scoreForm}>
+            <div className={styles.scores}>
+                {scoreStatsValue.map((stats) => (
+                    <KeyFigure
+                        key={stats.key}
+                        label={stats.label}
+                        value={stats.score}
+                    />
+                ))}
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx]: use ListView

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx (hunk 9 above)
Comment (shreeyash07) [reply app/views/EditAry/AssessmentRegistryForm/ScoreForm/index.tsx:129]: Not possible median returns undefined if array length is >= 0

**Review by AdityaKhatri — DISMISSED** [review]

**Pull request merged**